### PR TITLE
Feature/#13 animals

### LINKED ISSUE #13: Feature/#13 - Animals
Create classes of all animals

### PR DESCRIPTION
**Pull request by DominikNowak**
closes #13

### CODE REVIEW

File: src/Animals/Animal.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { AnimalNames } from '../Enums/AnimalNamesEnum';
+
+export abstract class Animal {
+  constructor(
+    protected name: AnimalNames,
+    protected value?: number,
```
Comment (vieraboschkova) [inline src/Animals/Animal.ts]: Is it optional? It is not necessary in wolf and fox case, is this correct?

File: src/Animals/BigDog.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { AnimalNames } from '../Enums/AnimalNamesEnum';
+import { ProtectHerdInterface } from '../Interfaces/ProtectHerdInterface';
+import { Animal } from './Animal';
+
+export class BigDog extends Animal implements ProtectHerdInterface {
+  constructor() {
+    super(AnimalNames.BIG_DOG, 36);
```
Comment (vieraboschkova) [inline src/Animals/BigDog.ts]: What are these numbers? 36, 72 ?

File: src/Animals/Animal.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  ) {}
+
+  get theName(): string {
+    return this.name;
+  }
+
+  get theValue(): number {
+    if (this.value) return this.value;
+    throw new TypeError(`Value of ${this.name} is undefined`);
```
Comment (vieraboschkova) [inline src/Animals/Animal.ts:32]: Consider changing:
`if(!this.value) throw... `
`return this.value`

File: src/Animals/Animal.ts (hunk 1 above)
Comment (Enessetere) [reply src/Animals/Animal.ts]: yes, value is optional, applies only to herd animals, like cow and rabbit. Predators like wolf and fox will not use this field.

File: src/Animals/BigDog.ts (hunk 2 above)
Comment (Enessetere) [reply src/Animals/BigDog.ts]: These numbers are value of Animal in lowest currency

**Review by Enessetere — CHANGES_REQUESTED** [review]
Looks good to me. Well done.

Merge with lastest develop and resolve conflicts.

File: src/Animals/BigDog.ts (hunk 2 above)
Comment (vieraboschkova) [reply src/Animals/BigDog.ts]: Animal Dollars :)

**Review by Enessetere — APPROVED** [review]
Looks good to me, well done.

**Review by SebastianBabinski1 — APPROVED** [review]
Well done!

**Pull request merged**